pkg/manager/client/discovery.go

```diff
@@ -30,35 +30,50 @@ func NewDiscoveryClient(ctx context.Context, config *rest.Config) (*DiscoveryCli
 	}, nil
 }
 
-func toObj(b []byte) interface{} {
-	var objMap map[string]interface{}
+func toObj(b []byte, groupVersion, kind string) (interface{}, error) {
 
-	r := bytes.NewReader(b)
-	d := json.NewDecoder(r)
+	replaceString := strings.ReplaceAll(string(b), `"creationTimestamp":null`, `"creationTimestamp":"null"`)
+	replaceString = strings.ReplaceAll(replaceString, `\"creationTimestamp\":null`, `\"creationTimestamp\":\"null\"`)
 
-	if err := d.Decode(&objMap); err != nil {
-		logrus.Errorf("Unable to decode json.")
-		return nil
+	finalString := strings.ReplaceAll(replaceString, `""`, `"null"`)
+	jsonParsed, err := gabs.ParseJSON([]byte(finalString))
```

Comment:
I just noticed. Why it's JSON format instead of yaml?

Reply:
The result is a byte slice which gets decode to json for further manipulation and then gets serialized  as yaml file here https://github.com/shangma/support-bundle-kit/blob/a422256672ef941438c37c1f347f9e7db38dd363/pkg/manager/cluster.go#L145

Reply:
Need to add comments in https://github.com/shangma/support-bundle-kit/blob/a422256672ef941438c37c1f347f9e7db38dd363/pkg/manager/cluster.go#L145 regarding why cannot we directly encode into YAML. I assume it's because of some of the options?

Reply:
Just did. @yasker Looks like no matter even yaml is setting to true in the option, still only a subset of it which is json compatible can be generated.